--- /opt/zenoss/Products/ZenUI3/browser/resources/js/zenoss/ComponentPanel.js	2012-08-01 18:24:40.000000000 -0400
+++ /opt/zenoss/Products/ZenUI3/browser/resources/js/zenoss/ComponentPanel.js	2012-08-23 07:53:43.443097331 -0400
@@ -565,9 +565,6 @@
             var token = Ext.History.getToken();
             if (token.split(Ext.History.DELIMITER).length!=3) {
                 this.getSelectionModel().selectRange(0, 0);
-                // Ext, for some reason, doesn't fire selectionchange at this
-                // point, so we'll do it ourselves.
-               this.fireEvent('selectionchange', this, this.getSelectionModel().getSelection());
             }
         }, this, {single:true});
         this.callParent(arguments);
