--- /opt/zenoss/Products/ZenWidgets/skins/zenui/javascript/geomap.js    2012-08-22 15:37:56.119985938 -0400
+++ /opt/zenoss/Products/ZenWidgets/skins/zenui/javascript/geomap.js    2012-08-23 08:46:55.154994124 -0400
@@ -16,6 +16,9 @@
         var markers = [];
         var polylineregistry = [];
         var dialog = null;
+        var nodedata = null;
+        var linkdata = null;
+        var errorCount = 0;
     
     /* PUBLICIZE */
     ZenGeoMapPortlet.prototype = {
@@ -90,6 +93,17 @@
                         content += '<img src="http://us.i1.yimg.com/us.yimg.com/i/us/per/gr/gp/rel_interstitial_loading.gif" />';
                         dialog.innerHTML = content;                 
                         _overlay.constructMarker(results, true);
+                    }else if(status === google.maps.GeocoderStatus.OVER_QUERY_LIMIT) {
+                        /*  try the address a few times after some delay to make sure it really is a query limit
+                            problem and not just erroring becuase we hit it too many times a second. We can get
+                            this error when the user has reached their daily limit for their IP as well.
+                        */
+                        errorCount++;
+                        if(errorCount >= 5){
+                            _utils.statusDialog("QUERY_LIMIT error. If this is a free account, you may have reached your daily limit. Please try again later.");
+                            return false;
+                        }
+                        setTimeout(function(){_overlay.addMarkers()}, 2000);
                     }else{
                         _utils.statusDialog(status);
                         dialog.style.display = 'block';
@@ -136,6 +150,18 @@
                                 _utils.statusDialog(status);
                             }
                         });                 
+                    }else if(status === google.maps.GeocoderStatus.OVER_QUERY_LIMIT) {
+                        /*  try the address a few times after some delay to make sure it really is a query limit
+                            problem and not just erroring becuase we hit it too many times a second. We can get
+                            this error when the user has reached their daily limit for their IP as well.
+                        */
+                        errorCount++;
+                        if(errorCount >= 5){
+                            _utils.statusDialog("QUERY_LIMIT error. If this is a free account, you may have reached your daily limit. Please try again later.");
+                            return false;
+                        }
+                        setTimeout(function(){_overlay.addPolyline()}, 2000);
                     }else{
                         _utils.statusDialog(status);
                     }
@@ -199,7 +225,7 @@
                 // done with that marker, but wait, there's more...
                 // need a delay here to keep google from saying: OVER_QUERY_LIMIT
                 // due to having too many queries per second
-                setTimeout(function(){_overlay.addMarkers()}, 200);
+                setTimeout(function(){_overlay.addMarkers()}, 400);
             }
                         
         },
@@ -229,7 +255,7 @@
                 return;
             }else{
                 // done with that line, but wait, there's more...
-                setTimeout(function(){_overlay.addPolyline()}, 200);                           
+                setTimeout(function(){_overlay.addPolyline()}, 400);                           
             }
         },
         doDraw: function(results) {      
